node/service/src/relay_chain_selection.rs

```diff
@@ -339,6 +344,11 @@ where
 				None => return Ok(Some(target_hash)),
 				Some(best) => best,
 			}
+		} else {
+			match best_leaf {
+				None => return Ok(Some(target_hash)),
+				Some(best_leaf) => best_leaf,
+			}
```

Comment:
@rphmeier could double check this one? The case where we did not get a best/longest chain in the first place. I think this is the best we can do in that case.

Reply:
Yeah, I also agree. We'll need to rethink when we re-enable chain selection